client/web/src/insights/pages/dashboards/dashboard-page/components/add-insight-modal/components/add-insight-modal-content/AddInsightModalContent.tsx

```diff
@@ -0,0 +1,102 @@
+import classnames from 'classnames'
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import { ErrorAlert } from '../../../../../../../../components/alerts'
+import { LoaderButton } from '../../../../../../../../components/LoaderButton'
+import { FormInput } from '../../../../../../../components/form/form-input/FormInput'
+import { useCheckboxes } from '../../../../../../../components/form/hooks/useCheckboxes'
+import { useField } from '../../../../../../../components/form/hooks/useField'
+import { SubmissionErrors, useForm, FORM_ERROR } from '../../../../../../../components/form/hooks/useForm'
+import { Badge } from '../../../dashboard-select/components/badge/Badge'
+import { TruncatedText } from '../../../dashboard-select/components/trancated-text/TrancatedText'
+import { ReachableInsight } from '../../hooks/use-reachable-insights'
+
+import styles from './AddInsightModalContent.module.scss'
+
+interface AddInsightModalContentProps {
+    insights: ReachableInsight[]
+    initialValues: AddInsightFormValues
+    onSubmit: (values: AddInsightFormValues) => SubmissionErrors | Promise<SubmissionErrors> | void
+    onCancel: () => void
+}
+
+export interface AddInsightFormValues {
+    searchInput: string
+    insightIds: string[]
+}
+
+export const AddInsightModalContent: React.FunctionComponent<AddInsightModalContentProps> = props => {
+    const { initialValues, insights, onSubmit, onCancel } = props
+
+    const { formAPI, ref, handleSubmit } = useForm({
+        initialValues,
+        onSubmit,
+    })
+
+    const searchInput = useField('searchInput', formAPI)
+    const {
+        input: { isChecked, onChange, onBlur },
+    } = useCheckboxes('insightIds', formAPI)
+
+    const filteredInsights = insights.filter(insight => insight.title.match(new RegExp(searchInput.input.value, 'gi')))
+
+    return (
+        // eslint-disable-next-line react/forbid-elements
+        <form ref={ref} onSubmit={handleSubmit}>
+            <FormInput
+                autoFocus={true}
+                title="Filter your insights"
+                description={
+                    <span className="">
+                        Couldn't find your insight? Check your insight's visibillty settings or{' '}
```

Comment:
```suggestion
                        Couldn't find your insight? Check your insight's visibility settings or{' '}
```

Reply:
```suggestion
                        Don't see an insight? Check the insight's visibility settings or{' '}
```
Just shorter, but your version was also good!